azure-pipelines.yml

```diff
@@ -0,0 +1,81 @@
+name: 3.0.0-ci.$(Date:yyyyMMdd).$(Rev:r) # equal to $(Build.BuildNumber)
+trigger:
+  branches:
+    include:
+    - dev
+    - master
+    - hotfix/*
+    - release/*
+variables:
+  DOTNET_CLI_TELEMETRY_OPTOUT: 1
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+  DOTNET_SDK_VERSION: '3.0.100-preview3-010431'
+jobs:
+- job: Linux
+  pool:
+    vmImage: ubuntu-16.04
+  steps:
+  - bash: sudo sh .build/setup_postgres.sh
+    displayName: 'Start PostgreSQL'
+  - task: DotNetCoreInstaller@0
+    displayName: 'Install .NET Core SDK'
+    inputs:
+      packageType: 'sdk'
+      version: '$(DOTNET_SDK_VERSION)'
+  - task: DotNetCoreCLI@2
+    displayName: 'Test'
+    inputs:
+      command: 'test'
+      configuration: 'Debug'
+      projects: 'test/**/*.csproj'
+      publishTestResults: true
+- job: Windows
+  pool:
+    vmImage: vs2017-win2016
+  steps:
+  - powershell: Write-Host "##vso[task.setvariable variable=truncated_sha1]$($env:BUILD_SOURCEVERSION.subString(0, 9))";
+    displayName: 'Assign truncated SHA1 to environment variable'
+  - powershell: .build/setup_postgres.ps1
+    displayName: 'Start PostgreSQL'
+  - task: DotNetCoreInstaller@0
+    displayName: 'Install .NET Core SDK'
+    inputs:
+      packageType: 'sdk'
+      version: '$(DOTNET_SDK_VERSION)'
+  - task: DotNetCoreCLI@2
+    displayName: 'Test'
+    inputs:
+      command: 'test'
+      configuration: 'Debug'
+      projects: 'test/**/*.csproj'
+      publishTestResults: true
+  - task: DotNetCoreCLI@2
+    displayName: 'Package'
+    inputs:
+      command: 'pack'
+      configuration: 'Release'
+      buildProperties: 'Version=$(Build.BuildNumber)+sha.$(truncated_sha1)'
+      packagesToPack: 'src/**/*.csproj'
+      packDirectory: '$(System.DefaultWorkingDirectory)'
+      verbosityPack: 'normal'
+  # TODO: needed because DotNetCoreCLI@2 doesn't support authenticated feeds yet.
+  # TODO: see https://github.com/Microsoft/azure-pipelines-tasks/issues/7160
+  - task: NuGetToolInstaller@0
```

Comment:
I think that as part of the move to Azure Pipelines, we would also move our two feeds away from myget. This would probably simplify the complexity here (see https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/nuget?view=azure-devops&tabs=yaml).

We can handle this a bit later though, after merging this PR.